kasa/deviceconfig.py

```diff
@@ -17,25 +17,32 @@
 >>> config_dict = device.config.to_dict()
 >>> # DeviceConfig.to_dict() can be used to store for later
 >>> print(config_dict)
-{'host': '127.0.0.3', 'timeout': 5, 'credentials': Credentials(), 'connection_type'\
-: {'device_family': 'SMART.TAPOBULB', 'encryption_type': 'KLAP', 'https': False, \
-'login_version': 2}, 'uses_http': True}
+{'host': '127.0.0.3', 'timeout': 5, 'credentials': {'username': 'user@example.com', \
+'password': 'great_password'}, 'connection_type'\
+: {'device_family': 'SMART.TAPOBULB', 'encryption_type': 'KLAP', 'login_version': 2, \
+'https': False}, 'uses_http': True}
```

Comment:
Identified a bug here whereby `credentials` were not actually being fully serialized previously.

Reply:
That's sort of expected with any custom (de)serialization code, so thanks a lot for doing the conversion! :-)